### PR TITLE
[Python3 migration]Fix test failure in dualtor and revert bad fixes during Python3 migration

### PR DESCRIPTION
**Pull request by wsycqyz**
### Description of PR

Summary:
Fixes test failures after Python3 migration in tests/dualtor directory.
dualtor.test_ipinip
dualtor.test_tor_ecn

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
Test failures after Python3 migration in tests/dualtor directory.
dualtor.test_ipinip
dualtor.test_tor_ecn

#### How did you do it?
1. Use six.text_type()
2. Use packet.show2() instead of packet.show() in Python3
3. Use pkt_match(packet[0]) instead of pkt_match(packet) in Python3
4. Search the code base, fix possible issue with ip_address() and decode().
5. Fix bad fix in first batch Python3 migration(#7748 )

#### How did you verify/test it?
Manually run test cases and got 100% pass.

### CODE REVIEW

File: tests/common/dualtor/dual_tor_utils.py [hunk 1]
```diff
@@ -929,7 +930,7 @@ def _generate_hashed_ipv6_packet(src_mac, dst_mac, dst_ip, hash_key):
     if len(generate_hashed_packet_to_server.packets_cache[call_signature]) < count:
         pkt_num = count - len(generate_hashed_packet_to_server.packets_cache[call_signature])
         for _ in range(pkt_num):
-            if ipaddress.ip_address(target_server_ip.decode()).version == 4:
+            if ipaddress.ip_address(six.text_type(target_server_ip)).version == 4:
```
Comment (wangxin) [inline tests/common/dualtor/dual_tor_utils.py:933]: Probably do a search in to code base to see if there is similar issue in other files?

Comment (wsycqyz) [reply tests/common/dualtor/dual_tor_utils.py:933]: Will do

Comment (wsycqyz) [reply tests/common/dualtor/dual_tor_utils.py:933]: Search code base and change another two places.
There are a few other code using ip_address() with decode() but fixed by Xiawei Jiang.

**Review by wangxin — APPROVED** [review]

**Pull request merged**